### ISSUE TITLE
POSIX dependency crashes meteor tool on Windows

### ISSUE THREAD

**Issue by buzzard192** [meteor/meteor#12940]
Installing or upgrading to the latest alpha (3.0-alpha.20) on Windows errors out due to POSIX attempting to be included in the webapp package.  

To fix the issue and successfully upgrade, remove the two files listed below from: 
     .meteor\packages\webapp\2.0.0-alpha300.20\npm\node_modules

- .npm-shrinkwrap.json
- .package-lock.json

also delete the POSIX module directory that was created during the installation attempt.

This issue is the same as the previous issue:
[POSIX dependency crashes meteor tool on Windows](https://github.com/meteor/meteor/issues/11497)

**Comment by jbo023** [comment]
We do have the same problem for Meteor 3.0 beta4,
```
While loading package webapp@2.0.0-beta300.4:
error: Command failed: C:\WINDOWS\system32\cmd.exe /c
C:\Users\jbo02\AppData\Local\.meteor\packages\meteor-tool\3.0.0-beta.4\mt-os.windows.x86_64\dev_bundle\bin\npm.cmd
rebuild --update-binary
npm ERR! code 1
npm ERR! path
C:\Users\jbo02\AppData\Local\.meteor\packages\webapp\2.0.0-beta300.4\npm\node_modules\.temp-14ktj0g.milg\node_modules\posix
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c node-gyp rebuild
npm ERR! posix.cc
npm ERR! C:\Users\jbo02\AppData\Local\.meteor\packages\webapp\2.0.0-beta300.4\npm\node_modules\nan\nan.h(688,37):
warning C4996: 'v8::Isolate::IdleNotificationDeadline': Use MemoryPressureNotification() to influence the GC
schedule.
[C:\Users\jbo02\AppData\Local\.meteor\packages\webapp\2.0.0-beta300.4\npm\node_modules\.temp-14ktj0g.milg\node_modules\posix\build\posix.vcxproj]
npm ERR!   (Quelldatei "../src/posix.cc" wird kompiliert)
npm ERR!
npm ERR!
C:\Users\jbo02\AppData\Local\.meteor\packages\webapp\2.0.0-beta300.4\npm\node_modules\.temp-14ktj0g.milg\node_modules\posix\src\posix.cc(2,10):
error C1083: Datei (Include) kann nicht ge�ffnet werden: "unistd.h": No such file or directory
[C:\Users\jbo02\AppData\Local\.meteor\packages\webapp\2.0.0-beta300.4\npm\node_modules\.temp-14ktj0g.milg\node_modules\posix\build\posix.vcxproj]
npm ERR! gyp ERR! build error
npm ERR! gyp ERR! stack Error: `C:\Program Files\Microsoft Visual
Studio\2022\Community\MSBuild\Current\Bin\MSBuild.exe` failed with exit code: 1
npm ERR! gyp ERR! stack     at ChildProcess.onExit
(C:\Users\jbo02\AppData\Local\.meteor\packages\meteor-tool\3.0.0-beta.4\mt-os.windows.x86_64\dev_bundle\lib\node_modules\npm\node_modules\node-gyp\lib\build.js:203:23)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:514:28)
npm ERR! gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:294:12)
npm ERR! gyp ERR! System Windows_NT 10.0.22631
npm ERR! gyp ERR! command
"C:\\Users\\jbo02\\AppData\\Local\\.meteor\\packages\\meteor-tool\\3.0.0-beta.4\\mt-os.windows.x86_64\\dev_bundle\\bin\\node.exe"
"C:\\Users\\jbo02\\AppData\\Local\\.meteor\\packages\\meteor-tool\\3.0.0-beta.4\\mt-os.windows.x86_64\\dev_bundle\\lib\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js"
"rebuild"
npm ERR! gyp ERR! cwd
C:\Users\jbo02\AppData\Local\.meteor\packages\webapp\2.0.0-beta300.4\npm\node_modules\.temp-14ktj0g.milg\node_modules\posix
npm ERR! gyp ERR! node -v v20.9.0
npm ERR! gyp ERR! node-gyp -v v9.4.0
npm ERR! gyp ERR! not ok
```

**Comment by jankapunkt** [comment]
This is the same issue under Linux , https://github.com/meteor/meteor/issues/12890

**Comment by nachocodoner** [comment]
Do you think we could do something in the Meteor core level to get rid of these issues?

[In this issue](https://github.com/meteor/meteor/issues/13007 ) we also have identified an issue that requires a full or partial clean on the Meteor installation to get rid of it. In that case we need to clean the package db catalog. We may plan like a command `meteor refresh` to clean package db catalog. Not sure if the issue mentioned here or others alike could also be added as part of the "refresh" process.

**Comment by jankapunkt** [comment]
@buzzard192 do you have by any chance some python installations, multiple python versions etc?

**Comment by jankapunkt** [comment]
By the way this morning I came back to my project with this issue (the fix proposed by @buzzard192 worked) and this time I got 

```shell
/home/jankapunkt/.meteor/packages/meteor-tool/.3.0.0-beta.0.8fge4e.g2d3++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/tools/fs/files.js:1248
        throw new Error("couldn't read entire resource");
```

A simple `meteor reset` fixed it, however I don't know how this happened as it run fine after the posix fix. Just wanted to add this here in case it might help in finding the issue's cause.

**Comment by zodern** [comment]
The problem is how npm rebuilds npm packages and how Meteor publishes packages is incompatible with optional dependencies that are normally only installed on certain os's. In this case, the posix dependency is an optional dependency, and would not normally be installed on Windows. However, since the webapp package was published from a different os, the posix dependency was included in the package.

Meteor detects packages that potentially have native dependencies, and have npm rebuild them after downloading the package. On Windows, rebuilding the package fails.

One solution might be for Meteor to also check for optional dependencies and rebuild them separately, ignoring errors.

However, this does not work for many newer packages like swc or esbuild that have a separate optional dependency for each platform/architecture: https://github.com/evanw/esbuild/blob/9d1777f23d9b64c186345223d92f319e59388d8b/npm/esbuild/package.json#L21-L43
Meteor would need to re-install `esbuild` instead of simply rebuilding it for the correct optional dependency to be installed.

I had worked around this to use swc in zodern:standard-minifier-js by creating a [wrapper package](https://github.com/zodern/meteor-package-install-swc) that installs swc in a post install script after Meteor downloads zodern:standard-minifier-js.

**Comment by mib00038** [comment]
Getting same issue on Ubuntu 22.04.3 LTS when creating new project --release 3.0-beta.6

also installing gyp-node did not resolve this.

```
Something has happened while creating your app using git clone.
Will use cached version of skeletons.
Error message:  Using release option
=> Errors while creating your project         
                                              
While loading package webapp@2.0.0-beta300.6:
error: Command failed: /home/mike/.meteor/packages/meteor-tool/.3.0.0-beta.6.1hgtlg3.yvcz++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/bin/npm
rebuild --update-binary
npm ERR! code 1
npm ERR! path /home/mike/.meteor/packages/webapp/.2.0.0-beta300.6.oo410c.an2sr++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/.temp-szxz5i.uhla/node_modules/posix
npm ERR! command failed
npm ERR! command sh -c node-gyp rebuild
npm ERR! make: Entering directory
'/home/mike/.meteor/packages/webapp/.2.0.0-beta300.6.oo410c.an2sr++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/.temp-szxz5i.uhla/node_modules/posix/build'
npm ERR!   CXX(target) Release/obj.target/posix/src/posix.o
npm ERR! make: Leaving directory
'/home/mike/.meteor/packages/webapp/.2.0.0-beta300.6.oo410c.an2sr++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/.temp-szxz5i.uhla/node_modules/posix/build'
npm ERR! make: g++: No such file or directory
npm ERR! make: *** [posix.target.mk:108: Release/obj.target/posix/src/posix.o] Error 127
npm ERR! gyp ERR! build error 
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack at ChildProcess.<anonymous>
(/home/mike/.meteor/packages/meteor-tool/.3.0.0-beta.6.1hgtlg3.yvcz++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:209:23)
npm ERR! gyp ERR! System Linux 6.5.0-26-generic
npm ERR! gyp ERR! command
"/home/mike/.meteor/packages/meteor-tool/.3.0.0-beta.6.1hgtlg3.yvcz++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/bin/node"
"/home/mike/.meteor/packages/meteor-tool/.3.0.0-beta.6.1hgtlg3.yvcz++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js"
"rebuild"
npm ERR! gyp ERR! cwd /home/mike/.meteor/packages/webapp/.2.0.0-beta300.6.oo410c.an2sr++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/.temp-szxz5i.uhla/node_modules/posix
npm ERR! gyp ERR! node -v v20.11.1
npm ERR! gyp ERR! node-gyp -v v10.0.1
npm ERR! gyp ERR! not ok

npm ERR! A complete log of this run can be found in: /home/mike/.npm/_logs/2024-03-22T14_25_35_091Z-debug-0.log
npm ERR! code 1
npm ERR! path /home/mike/.meteor/packages/webapp/.2.0.0-beta300.6.oo410c.an2sr++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/.temp-szxz5i.uhla/node_modules/posix
npm ERR! command failed
npm ERR! command sh -c node-gyp rebuild
npm ERR! make: Entering directory
'/home/mike/.meteor/packages/webapp/.2.0.0-beta300.6.oo410c.an2sr++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/.temp-szxz5i.uhla/node_modules/posix/build'
npm ERR!   CXX(target) Release/obj.target/posix/src/posix.o
npm ERR! make: Leaving directory
'/home/mike/.meteor/packages/webapp/.2.0.0-beta300.6.oo410c.an2sr++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/.temp-szxz5i.uhla/node_modules/posix/build'
npm ERR! make: g++: No such file or directory
npm ERR! make: *** [posix.target.mk:108: Release/obj.target/posix/src/posix.o] Error 127
npm ERR! gyp ERR! build error 
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack at ChildProcess.<anonymous>
(/home/mike/.meteor/packages/meteor-tool/.3.0.0-beta.6.1hgtlg3.yvcz++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:209:23)
npm ERR! gyp ERR! System Linux 6.5.0-26-generic
npm ERR! gyp ERR! command
"/home/mike/.meteor/packages/meteor-tool/.3.0.0-beta.6.1hgtlg3.yvcz++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/bin/node"
"/home/mike/.meteor/packages/meteor-tool/.3.0.0-beta.6.1hgtlg3.yvcz++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js"
"rebuild"
npm ERR! gyp ERR! cwd /home/mike/.meteor/packages/webapp/.2.0.0-beta300.6.oo410c.an2sr++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/.temp-szxz5i.uhla/node_modules/posix
npm ERR! gyp ERR! node -v v20.11.1
npm ERR! gyp ERR! node-gyp -v v10.0.1
npm ERR! gyp ERR! not ok
```

**Comment by jankapunkt** [comment]
Interestingly this happens only on one of my machines both running xubuntu 22.04. and I wonder what's causing this on one machine but not on the other : 🤔

The only difference I know so far is the one machine with the issues has multiple python versions installed.

**Comment by mib00038** [comment]
@jankapunkt are both machines show same version when running command `meteor --version` outside of a project ?

**Comment by jankapunkt** [comment]
Outside a project both are on 2.15

**Comment by jbo023** [comment]
The problem still persists with Meteor 3.0 Beta7, workaround still works.

**Comment by milanzigmond** [comment]
I have the same issue on beta7 on windows but do not understand how to fix it, where exactly do I remove those files? do I need to make local copy or webapp beta7? I'm just confused about the fix, can someone explain please, or better fix it? thanks

**Comment by StorytellerCZ** [comment]
@nachocodoner could this be related to #13090 issues you are facing?

**Comment by jbo023** [comment]
@milanzigmond  you can fix it by doing the following:

In the Folder
AppData\Local\.meteor\packages\webapp\2.0.0-alpha300.20\npm\node_modules

remove the following Files:

.npm-shrinkwrap.json
.package-lock.json
also delete the POSIX module directory that was created during the installation attempt.


The `2.0.0-alpha300.20` can now also be named different, thats probably an old webapp version.

**Comment by nachocodoner** [comment]
> could this be related to https://github.com/meteor/meteor/pull/13090 issues you are facing?

Not related. The issue there is about the upgrade of Node.js to latest version and a breaking scenario on Windows executable. But not affecting on beta.7. The issue here is about caching on incremental builds.

**Comment by milanzigmond** [comment]
Thank you @jbo023!

I didn't get that files need to be removed from the GLOBAL meteor installation, I was looking inside a single project folder :).

**Comment by ruohki** [comment]
This is still relevant for the RC.0

**Comment by jbo023** [comment]
yeah unfortunatly see https://forums.meteor.com/t/meteor-3-0-rc-0-is-out/61515/26

**Comment by ruohki** [comment]
The fix in this issue helps, I just want to make sure not be left out, like with the performance issues on Windows. At least it should work, I don't really care for slow.

**Comment by ruohki** [comment]
Can't we just add a post install script to the installer or something that simply purges the POSIX module from the webapp when we are on the windows platform?

**Comment by jankapunkt** [comment]
Just want to let you know this is still an issue for me on Ubuntu with 3.0 RC.4

**Comment by etulikov** [comment]
Same issue on MacOS.

```
While loading package webapp@2.0.0-rc300.4:
   error: Command failed:
   /Users/etulikov/.meteor/packages/meteor-tool/.3.0.0-rc.4.em9tx5.5iy2t++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/bin/npm rebuild
   --update-binary
   npm ERR! code 1
   npm ERR! path
   /Users/etulikov/.meteor/packages/webapp/.2.0.0-rc300.4.cs3djb.ohk5e++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/.temp-1pildq5.x9vo/node_modules/posix
   npm ERR! command failed
   npm ERR! command sh -c node-gyp rebuild
   npm ERR! CXX(target) Release/obj.target/posix/src/posix.o
   npm ERR! In file included from ../src/posix.cc:1:
   npm ERR! In file included from ../../../../nan/nan.h:61:
   npm ERR! In file included from /Users/etulikov/Library/Caches/node-gyp/20.12.2/include/node/uv.h:56:
   npm ERR! /Users/etulikov/Library/Caches/node-gyp/20.12.2/include/node/uv/errno.h:25:10: error: 'errno.h' file not found with <angled> include; use "quotes" instead
   npm ERR! #include <errno.h>
   npm ERR!          ^~~~~~~~~
   npm ERR!          "errno.h"
   npm ERR! In file included from ../src/posix.cc:1:
   npm ERR! In file included from ../../../../nan/nan.h:61:
   npm ERR! /Users/etulikov/Library/Caches/node-gyp/20.12.2/include/node/uv.h:59:10: fatal error: 'stdio.h' file not found
   npm ERR! #include <stdio.h>
   npm ERR!          ^~~~~~~~~
   npm ERR! 2 errors generated.
   npm ERR! make: *** [Release/obj.target/posix/src/posix.o] Error 1
   npm ERR! gyp ERR! build error
   npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
   npm ERR! gyp ERR! stack at ChildProcess.<anonymous>
   (/Users/etulikov/.meteor/packages/meteor-tool/.3.0.0-rc.4.em9tx5.5iy2t++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:209:23)
   npm ERR! gyp ERR! System Darwin 23.5.0
   npm ERR! gyp ERR! command
   "/Users/etulikov/.meteor/packages/meteor-tool/.3.0.0-rc.4.em9tx5.5iy2t++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/bin/node"
   "/Users/etulikov/.meteor/packages/meteor-tool/.3.0.0-rc.4.em9tx5.5iy2t++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js"
   "rebuild"
   npm ERR! gyp ERR! cwd
   /Users/etulikov/.meteor/packages/webapp/.2.0.0-rc300.4.cs3djb.ohk5e++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/.temp-1pildq5.x9vo/node_modules/posix
   npm ERR! gyp ERR! node -v v20.12.2
   npm ERR! gyp ERR! node-gyp -v v10.0.1
   npm ERR! gyp ERR! not ok

   npm ERR! A complete log of this run can be found in: /Users/etulikov/.npm/_logs/2024-06-24T19_24_22_809Z-debug-0.log
   npm ERR! code 1
   npm ERR! path
   /Users/etulikov/.meteor/packages/webapp/.2.0.0-rc300.4.cs3djb.ohk5e++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/.temp-1pildq5.x9vo/node_modules/posix
   npm ERR! command failed
   npm ERR! command sh -c node-gyp rebuild
   npm ERR! CXX(target) Release/obj.target/posix/src/posix.o
   npm ERR! In file included from ../src/posix.cc:1:
   npm ERR! In file included from ../../../../nan/nan.h:61:
   npm ERR! In file included from /Users/etulikov/Library/Caches/node-gyp/20.12.2/include/node/uv.h:56:
   npm ERR! /Users/etulikov/Library/Caches/node-gyp/20.12.2/include/node/uv/errno.h:25:10: error: 'errno.h' file not found with <angled> include; use "quotes" instead
   npm ERR! #include <errno.h>
   npm ERR!          ^~~~~~~~~
   npm ERR!          "errno.h"
   npm ERR! In file included from ../src/posix.cc:1:
   npm ERR! In file included from ../../../../nan/nan.h:61:
   npm ERR! /Users/etulikov/Library/Caches/node-gyp/20.12.2/include/node/uv.h:59:10: fatal error: 'stdio.h' file not found
   npm ERR! #include <stdio.h>
   npm ERR!          ^~~~~~~~~
   npm ERR! 2 errors generated.
   npm ERR! make: *** [Release/obj.target/posix/src/posix.o] Error 1
   npm ERR! gyp ERR! build error
   npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
   npm ERR! gyp ERR! stack at ChildProcess.<anonymous>
   (/Users/etulikov/.meteor/packages/meteor-tool/.3.0.0-rc.4.em9tx5.5iy2t++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:209:23)
   npm ERR! gyp ERR! System Darwin 23.5.0
   npm ERR! gyp ERR! command
   "/Users/etulikov/.meteor/packages/meteor-tool/.3.0.0-rc.4.em9tx5.5iy2t++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/bin/node"
   "/Users/etulikov/.meteor/packages/meteor-tool/.3.0.0-rc.4.em9tx5.5iy2t++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js"
   "rebuild"
   npm ERR! gyp ERR! cwd
   /Users/etulikov/.meteor/packages/webapp/.2.0.0-rc300.4.cs3djb.ohk5e++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/.temp-1pildq5.x9vo/node_modules/posix
   npm ERR! gyp ERR! node -v v20.12.2
   npm ERR! gyp ERR! node-gyp -v v10.0.1
   npm ERR! gyp ERR! not ok

   npm ERR! A complete log of this run can be found in: /Users/etulikov/.npm/_logs/2024-06-24T19_24_22_809Z-debug-0.log
```

**Comment by pmogollons** [comment]
Same issue on ubuntu 24.04

```bash
=> Errors while initializing project:                                              
                                              
While loading webapp@2.0.0-rc300.4...:
error: Command failed:
/home/wanchope/.meteor/packages/meteor-tool/.3.0.0-rc.4.15dcwo5.4pd3k++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/bin/npm
rebuild --update-binary
npm ERR! code 1
npm ERR! path /tmp/mt-15ei3ux.f4db/npm/node_modules/.temp-1d5bdvl.w0j5i/node_modules/posix
npm ERR! command failed
npm ERR! command sh -c node-gyp rebuild
npm ERR! make: Entering directory '/tmp/mt-15ei3ux.f4db/npm/node_modules/.temp-1d5bdvl.w0j5i/node_modules/posix/build'
npm ERR!   CXX(target) Release/obj.target/posix/src/posix.o
npm ERR! make: Leaving directory '/tmp/mt-15ei3ux.f4db/npm/node_modules/.temp-1d5bdvl.w0j5i/node_modules/posix/build'
npm ERR! In file included from ../../../../nan/nan.h:62:0,
npm ERR!                  from ../src/posix.cc:1:
npm ERR! /home/wanchope/.cache/node-gyp/20.12.2/include/node/node.h:83:20: fatal error: optional: No such file or directory
npm ERR! compilation terminated.
npm ERR! make: *** [posix.target.mk:108: Release/obj.target/posix/src/posix.o] Error 1
npm ERR! gyp ERR! build error 
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack at ChildProcess.<anonymous>
(/home/wanchope/.meteor/packages/meteor-tool/.3.0.0-rc.4.15dcwo5.4pd3k++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:209:23)
npm ERR! gyp ERR! System Linux 6.5.0-35-generic
npm ERR! gyp ERR! command
"/home/wanchope/.meteor/packages/meteor-tool/.3.0.0-rc.4.15dcwo5.4pd3k++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/bin/node"
"/home/wanchope/.meteor/packages/meteor-tool/.3.0.0-rc.4.15dcwo5.4pd3k++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js"
"rebuild"
npm ERR! gyp ERR! cwd /tmp/mt-15ei3ux.f4db/npm/node_modules/.temp-1d5bdvl.w0j5i/node_modules/posix
npm ERR! gyp ERR! node -v v20.12.2
npm ERR! gyp ERR! node-gyp -v v10.0.1
npm ERR! gyp ERR! not ok

npm ERR! A complete log of this run can be found in: /home/wanchope/.npm/_logs/2024-07-09T23_54_34_640Z-debug-0.log
npm ERR! code 1
npm ERR! path /tmp/mt-15ei3ux.f4db/npm/node_modules/.temp-1d5bdvl.w0j5i/node_modules/posix
npm ERR! command failed
npm ERR! command sh -c node-gyp rebuild
npm ERR! make: Entering directory '/tmp/mt-15ei3ux.f4db/npm/node_modules/.temp-1d5bdvl.w0j5i/node_modules/posix/build'
npm ERR!   CXX(target) Release/obj.target/posix/src/posix.o
npm ERR! make: Leaving directory '/tmp/mt-15ei3ux.f4db/npm/node_modules/.temp-1d5bdvl.w0j5i/node_modules/posix/build'
npm ERR! In file included from ../../../../nan/nan.h:62:0,
npm ERR!                  from ../src/posix.cc:1:
npm ERR! /home/wanchope/.cache/node-gyp/20.12.2/include/node/node.h:83:20: fatal error: optional: No such file or directory
npm ERR! compilation terminated.
npm ERR! make: *** [posix.target.mk:108: Release/obj.target/posix/src/posix.o] Error 1
npm ERR! gyp ERR! build error 
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack at ChildProcess.<anonymous>
(/home/wanchope/.meteor/packages/meteor-tool/.3.0.0-rc.4.15dcwo5.4pd3k++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:209:23)
npm ERR! gyp ERR! System Linux 6.5.0-35-generic
npm ERR! gyp ERR! command
"/home/wanchope/.meteor/packages/meteor-tool/.3.0.0-rc.4.15dcwo5.4pd3k++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/bin/node"
"/home/wanchope/.meteor/packages/meteor-tool/.3.0.0-rc.4.15dcwo5.4pd3k++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js"
"rebuild"
npm ERR! gyp ERR! cwd /tmp/mt-15ei3ux.f4db/npm/node_modules/.temp-1d5bdvl.w0j5i/node_modules/posix
npm ERR! gyp ERR! node -v v20.12.2
npm ERR! gyp ERR! node-gyp -v v10.0.1
npm ERR! gyp ERR! not ok

npm ERR! A complete log of this run can be found in: /home/wanchope/.npm/_logs/2024-07-09T23_54_34_640Z-debug-0.log
```

**Comment by nachocodoner** [comment]
This fix is scheduled to be released on next official version, https://github.com/meteor/meteor/pull/13204

Hopefully this will make this issue to not trigger anymore in any way.

**Comment by denihs** [comment]
This should be fixed. If you still see the issue, we can reopen it!

**Comment by jankapunkt** [comment]
At least from my end I can confirm this to be fixed on Ubuntu 22.04. LTS with Meteor 3.0.1 tested

**Comment by ruohki** [comment]
> At least from my end I can confirm this to be fixed on Ubuntu 22.04. LTS with Meteor 3.0.1 tested

Because this was a windows related issue :)

**Comment by jankapunkt** [comment]
It wasn't limited to windows and I commented a few times here about this happening on Ubuntu as well.
